### ISSUE TITLE
Permissive License - ASL 2.0 | MIT | BSD

### ISSUE THREAD

**Issue by sirinath** [Spreads/Spreads#6]
Can you consider a more permissive license like: ASL 2.0 | MIT | BSD

**Comment by buybackoff** [comment]
Nope. We do not have a big corporate sponsor and did this project for solving problems, not for PR or cross-sales, and hope that anyone who have the same problems will contribute - the library is quite small and self-contained. Also, we (will) likely depend on modified NetMQ. Finally, see [the story about Patrick](http://hintjens.com/blog:27) by @hintjens. ZeroMQ/NetMQ case shows that LGPL is well balanced, it is not (A)GPL and allows usage without restrictions, it is not viral if you do not change it but link to it.

Why would you want to use the code without syncing to the upstream? This kind of code - core low-level calculations primitives - benefits the most from wide adoption and community feedback.

**Comment by sirinath** [comment]
Also see: https://github.com/PerfectlySoft/Perfect/issues/131

**Comment by buybackoff** [comment]
TL;DR;...
Well, you have tons of other permissive FOSS projects to choose from. The license issue is not something to discuss. If you like this project so much, we could negotiate a commercial option (however, this is not something I am ready to spend much time). Otherwise it sounds like you just want some free lunch (especially given that you are pinging almost every other project on GitHub about such an issue). Currently I care about end users, and they do not care which license the project has.

**Comment by sirinath** [comment]
We are a small outfit and license matters to us. Say if we do a POC using a lib less number of clients have in their approved license list we cannot get a sale or work.

**Comment by hintjens** [comment]
Fwiw we've found that MPLv2 gives us the same benefits with less angst for
corporate users.

**Comment by buybackoff** [comment]
@hintjens yes, I am thinking about MPL2 after learning about it from ZeroMQ and related projects. I just do not understand all subtleties yet, while GPL is well understood, safe and default choice.

**Comment by sirinath** [comment]
Issue here is if we have to inevitably modify a file due to a bug, quick fix or workaround work paid by a client then this might not be the client's copyrights or governed by our contractual obligations. MPL is OK for internal use not for contractors and outsourcing.

**Comment by sirinath** [comment]
ASL 2.0, MIT and BSD are the safest bet that any potential client will have whitelisted hence safest to use in POC.

**Comment by hintjens** [comment]
One should always as a matter of principle get permission to modify
pre-existing open source works as a contractor, if such projects are the
basis for a work. And then to republish such patches under the existing
open source license, and under the name of the contractor. This removes the
corporate fear of liability.

**Comment by sirinath** [comment]
The above issue is not applicable to licenses like ASL 2.0, MIT and BSD. Also getting permissions is a overhead especially for small outfits. So most beneficial is to small time users are licenses like ASL 2.0, MIT and BSD. Also they are whitelisted by every existing and potential client.

**Comment by hintjens** [comment]
Even with those licenses a contractor would be forced to make and maintain
a dark fork without permission to submit patches.

**Comment by sirinath** [comment]
The overhead I meant is the negotiation overhead and cost if layers get involved. 

You only have to maintain a fork if a dependent project gets abandoned or certain blocking bugs do not get fixes which is not that often but if it inevitably happens you need to be covered.

I have never encountered having to take any special permission if all dependencies are permissive.

**Comment by sirinath** [comment]
We now have an internal project with does something similar for Scala but still we prefer ASL 2.0 as if we port any code which is under MPL then it should be MPL.
